### ISSUE TITLE
Renaming sideEffect() to observe() and introducing observeOnce()

### ISSUE THREAD

**Issue by chauthai** [mweststrate/mobservable#28]
During my discussion with @mweststrate of changing the sideEffect() api, we came up with two use cases which can benefit from observeOnce():

1. Notification System
You can use mobservable.observeOnce(func) to send a message to the server once the recently created instance of a notification is processed by the client. You would be able to set the notification state to `processed=true` once the information in the notification is processed.

2. Checkout Cart
Once the user payment is processed, the cart is freezed and can’t be rewinded like before the payment.

##### Old api approach
```javascript
var dispose = sideEffect(() => {
    if (notification.processed) {
         doServerThing()
         dispose()
   }
});
```
##### Suggested api changes by @mweststrate:
```javascript
once(() => notification.processed), () => doServerThing())`

````
or to conform with a Promise-like syntax:
```javascript
when(() => notification.processed).then(() => doServerThing())`
```

**Comment by mweststrate** [comment]
I just added `observe` as alias for `sideEffect`. `sideEffect` will be deprecated in a later version when all the docs are updated.

I also introduced `when` (currently undocumented). `when` has the following signature: 

```typescript
when(predicate: () => boolean, effect: () => void, scope?: any): () => void
```
`when` will observe predicate _until_ it returns true. Once `predicate` returns `true`, `effect` will be invoked (but only once).
`when` returns a function which can be used to end the watch on the predicate prematurely.

Using `when` in combination with promises could be as simple as:

```javascript
var promise = new Promise(function(resolver) {
      when(somePredicate, resolver);
});
```

`when` has been released as part of the 0.6.8 version, but is further undocumented. So please let me know what you think of the name and behavior :).

**Comment by mweststrate** [comment]
Ok, I think the proper name is `observeUntil`, a bit similar to RxJs' `takeUntil`. 

I'm also thinking of introducing another utility observer: `observeAsync`:

`observeAsync<T>(observer: () => T, effect: (lastValue: T) => void`.

Once the `observer` triggers, `effect` will be scheduled. If observer triggers multiple times, `effect` will still be triggered only once, so it achieves a similar effect as `transaction`. This might be useful for stuff that is expensive and doesn't need to happen synchronously; such as server communication.

**Comment by mweststrate** [comment]
All these features have been released as part of 0.7.
